File: code/eprogramoflifeapp/app/src/main/java/domain/MonthlyCommitment.java
package domain;

import java.text.SimpleDateFormat;
import java.util.Date;

/**
 * Created by mmfsf on 3/29/17.
 */

public class MonthlyCommitment extends Commitment {

    public MonthlyCommitment(String name) {
        super(name);
        this.frequency = Frequency.Monthly;
        simpleDateFormat = new SimpleDateFormat("yyyyMM");
    }

    public Level getPerformed(Date date)
    {
        String key = simpleDateFormat.format(date);

        Level level = this.performed.get(key);
        if(level == null)
        {
            level = Level.NotDone;
        }
        return level;
    }

    public void Point(Date date, Level level)
    {
        String key = simpleDateFormat.format(date);
        this.performed.put(key, level);
    }
}

File: code/eprogramoflifeweb/epl.core/Domain/Commitments/MonthlyCommitment.cs
using epl.core.ValuesObjects;
using System;

namespace epl.core.Domain
{
    public class MonthlyCommitment : Commitment
    {
        protected override string KeyFormat { get => "yyyyMM"; }

        public MonthlyCommitment(string name) : base(name)
        {
            Frequency = Frequency.Monthly;
        }

        public MonthlyCommitment(string name, string description) : base(name, description)
        {
            Frequency = Frequency.Monthly;
        }

        public override Level GetPoint(DateTime date)
        {
            var key = date.Date.ToString(this.KeyFormat);
            return Performed.ContainsKey(key) ? this.Performed[key] : Level.NotDone;
        }

        public override void Point(DateTime date, Level level)
        {
            var key = date.Date.ToString(this.KeyFormat);
            if (Performed.ContainsKey(key))
            {
                Performed[key] = level;
            }
            else
            {
                Performed.Add(key, level);
            }
        }
    }
}
